runners/samza/src/main/java/org/apache/beam/runners/samza/translation/GroupByKeyTranslator.java

```diff
@@ -112,7 +112,21 @@ public void translatePortable(
       PipelineNode.PTransformNode transform,
       QueryablePipeline pipeline,
       PortableTranslationContext ctx) {
-    doTranslatePortable(transform, pipeline, ctx);
+    final String inputId = ctx.getInputId(transform);
```

Comment:
why do we change the pattern to add impl in translatePortable instead of doTranslatePortable?

Reply:
Oh, we are reusing the doTranslatePortable() in the side input case to add GBK. let me put a java doc there.